src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -148,29 +164,59 @@ public RoutingResponse route(Router router) {
         this.debugAggregator.finishedPatternFiltering();
 
         // Prepare access/egress transfers
-        Collection<NearbyStop> accessStops = AccessEgressRouter.streetSearch(request, false, 2000);
-        Collection<NearbyStop> egressStops = AccessEgressRouter.streetSearch(request, true, 2000);
+        Collection<NearbyStop> accessStops = AccessEgressRouter.streetSearch(
+            request,
+            StreetMode.FLEXIBLE.equals(request.modes.accessMode)
+                ? StreetMode.WALK
+                : request.modes.accessMode,
+            false,
+            2000
+        );
+        Collection<NearbyStop> egressStops = AccessEgressRouter.streetSearch(
+            request,
+            StreetMode.FLEXIBLE.equals(request.modes.egressMode)
+                ? StreetMode.WALK
+                : request.modes.egressMode,
+            true,
+            2000
+        );
 
         AccessEgressMapper accessEgressMapper = new AccessEgressMapper(transitLayer.getStopIndex());
-        Collection<AccessEgress> accessTransfers = accessEgressMapper.mapNearbyStops(accessStops, false);
-        Collection<AccessEgress> egressTransfers = accessEgressMapper.mapNearbyStops(egressStops, true);
+        Collection<AccessEgress> accessTransfers = new ArrayList<>();
+        Collection<AccessEgress> egressTransfers = new ArrayList<>();
```

Comment:
`accessTransfers` is not a good name, maybe `accessListTransit` ?

Reply:
I have renamed them to accessList and egressList, but it is hard to find a name that fits.